tests/rule_based_profiler/parameter_builder/test_regex_pattern_string_parameter_builder.py

```diff
@@ -0,0 +1,233 @@
+from typing import Set
+
+import pytest
+
+import great_expectations.exceptions.exceptions as ge_exceptions
+from great_expectations.data_context import DataContext
+from great_expectations.execution_engine.execution_engine import MetricDomainTypes
+from great_expectations.rule_based_profiler.parameter_builder import (
+    RegexPatternStringParameterBuilder,
+    regex_pattern_string_parameter_builder,
+)
+from great_expectations.rule_based_profiler.types import (
+    Domain,
+    ParameterContainer,
+    get_parameter_value_by_fully_qualified_parameter_name,
+)
+
+
+def test_regex_pattern_string_parameter_builder_instantiation():
+    candidate_regexes: Set[str] = {
+        r"/\d+/",  # whole number with 1 or more digits ExpectValuesToBeNumeric? (.. youw oudl want to emit that expectation)?
+        r"/-?\d+/",  # negative whole numbers
+        r"/-?\d+(\.\d*)?/",  # decimal numbers with . (period) separator
+        r"/[A-Za-z0-9\.,;:!?()\"'%\-]+/",  # general text
+        r"^ +/",  # leading space
+        r" +/$",  # trailing space
+        r"/https?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#()?&//=]*)/",  # Matching URL (including http(s) protocol)
+        r"/<\/?(?:p|a|b|img)(?: \/)?>/",  # HTML tags
+        r"/(?:25[0-5]|2[0-4]\d|[01]\d{2}|\d{1,2})(?:.(?:25[0-5]|2[0-4]\d|[01]\d{2}|\d{1,2})){3}/",  # IPv4 IP address
+        r"/(?:[A-Fa-f0-9]){0,4}(?: ?:? ?(?:[A-Fa-f0-9]){0,4}){0,7}/",  # IPv6 IP address,
+    }
+    regex_pattern_string_parameter: RegexPatternStringParameterBuilder = (
+        RegexPatternStringParameterBuilder(
+            name="my_simple_regex_string_parameter_builder",
+            candidate_regexes=candidate_regexes,
+        )
+    )
+
+    assert regex_pattern_string_parameter._threshold == 0.9
+    assert regex_pattern_string_parameter._candidate_regexes == candidate_regexes
+    assert regex_pattern_string_parameter.CANDIDATE_REGEX == candidate_regexes
+
+
+def test_regex_pattern_string_parameter_builder_zero_batch_id_error():
+    regex_pattern_string_parameter: RegexPatternStringParameterBuilder = (
+        RegexPatternStringParameterBuilder(
+            name="my_simple_regex_string_parameter_builder",
+        )
+    )
+    parameter_container: ParameterContainer = ParameterContainer(parameter_nodes=None)
+    domain: Domain = Domain(domain_type=MetricDomainTypes.COLUMN)
+
+    with pytest.raises(ge_exceptions.ProfilerExecutionError) as e:
+        regex_pattern_string_parameter.build_parameters(
+            parameter_container=parameter_container, domain=domain
+        )
+
+    assert (
+        str(e.value)
+        == "RegexPatternStringParameterBuilder was not able to get Validator using domain, variables and parameters provided."
+    )
+
+
+def test_regex_pattern_string_parameter_builder_alice(
+    alice_columnar_table_single_batch_context,
+):
```

Comment:
I still somewhat hold the opinion that these fixtures constitute a unit test. I won't block this PR on that idea but I'd love to discuss this with the team and especially get @donaldheppner's point of view.